python-interface/src/MiniBotScripts/SwarmMaster.py

```diff
@@ -19,11 +20,102 @@ def run(bot):
     mediateThread = Thread(target=z.mediate)
     mediateThread.start()
     threads.append(mediateThread)
-    # commands for bot movement itself
+    
+    #echobot(bot,z)
+    colorbot(bot,z)
+    
+def colorbot(bot,z):
+    speed = 10
+    cs = bot.get_sensor_by_name("ColorSensor")
+    cs.calibrate()
+    pinkFirstTime = True
+    orangeFirstTime = True
+    count = {"F":0,"B":0,"L":0,"R":0}
+
+    try:
+        while(True):
+            c = cs.read_color()
+            if(c=="RED"):
+                # stop
+                msg = (0,0)
+                count["F"]=0
+                count["B"]=0
+                count["L"]=0
+                count["R"]=0
+                speed = 10
+            elif(c=="GREEN"):
+                # forwards
+                count["F"]+=1
+                count["B"]=0
+                count["L"]=0
+                count["R"]=0
+                if(count["F"]>15):
+                    count["F"]=0
+                    speed = inc(speed,15)
+                msg = (speed,speed)
+            elif(c=="BLUE"):
+                # backwards
+                count["F"]=0
+                count["B"]+=1
+                count["L"]=0
+                count["R"]=0
+                if(count["B"]>15):
+                    count["B"]=0
+                    speed = inc(speed,15)
+                msg = (-speed,-speed)
+            elif(c=="YELLOW"):
+                # turn left
+                count["F"]=0
+                count["B"]=0
+                count["L"]+=1
+                count["R"]=0
+                if(count["L"]>15):
+                    count["L"]=0
+                    speed = inc(speed,15)
+                msg = (-speed,speed)
+            elif(c=="VIOLET"):
+                # turn right
+                count["F"]=0
+                count["B"]=0
+                count["L"]=0
+                count["R"]+=1
+                if(count["R"]>15):
+                    count["R"]=0
+                    speed = inc(speed,15)
+                msg = (speed,-speed)
+            # elif(c=="PINK"):
+            #     # decrease speed
+            #     orangeFirstTime = True
+            #     if (pinkFirstTime && speed > 10):
+            #         pinkFirstTime = False
+            #         speed -= 5
+            #         print "SLOWER! (Speed: " + str(speed) + ")"
+            # elif(c=="ORANGE"):
+            #     # increase speed
+            #     pinkFirstTime = True
+            #     if (orangeFirstTime && speed < 50):
+            #         orangeFirstTime = False
+            #         speed += 5
+            #         print "FASTER! (Speed: " + str(speed) + ")"
+            
+            # print str(msg)
+            z.broadcast(msg)
+            time.sleep(0.2)
+    finally:
+        cleanup(z)
+
+def inc(speed, i):
+    if(speed<50):
+        speed += i
```

Comment:
this is fine, but I think it should be safe to just send a value > 50 since you are limiting elsewhere.